influxdb3/tests/server/auth.rs

```diff
@@ -144,13 +99,101 @@ async fn auth() {
+        .spawn()
+        .await;
+
+    // Write some data to the server, this will be authorized through the HTTP API
+    server
+        .write_lp_to_db(
+            "foo",
+            "cpu,host=s1,region=us-east usage=0.9 1\n\
+            cpu,host=s1,region=us-east usage=0.89 2\n\
+            cpu,host=s1,region=us-east usage=0.85 3",
+            Precision::Nanosecond,
+        )
+        .await
+        .unwrap();
+
+    {
+        // Spin up a FlightSQL client
+        let mut client = server.flight_sql_client("foo").await;
+
+        // Make a query using the client, without providing any authorization header:
+        let error = client.query("SELECT * FROM cpu").await.unwrap_err();
+        assert!(matches!(error, FlightError::Tonic(s) if s.code() == tonic::Code::Unauthenticated));
+
+        // Set the authorization header on the client:
+        client
+            .add_header("authorization", &format!("Bearer {TOKEN}"))
+            .unwrap();
+
+        // Make the query again, this time it should work:
+        let response = client.query("SELECT * FROM cpu").await.unwrap();
+        let batches = collect_stream(response).await;
+        assert_batches_sorted_eq!(
+            [
+                "+------+---------+--------------------------------+-------+",
+                "| host | region  | time                           | usage |",
+                "+------+---------+--------------------------------+-------+",
+                "| s1   | us-east | 1970-01-01T00:00:00.000000001Z | 0.9   |",
+                "| s1   | us-east | 1970-01-01T00:00:00.000000002Z | 0.89  |",
+                "| s1   | us-east | 1970-01-01T00:00:00.000000003Z | 0.85  |",
+                "+------+---------+--------------------------------+-------+",
+            ],
+            &batches
+        );
+    }
+
+    // Create some new clients that set the authorization header incorrectly to
+    // ensure errors are returned:
+
+    // Mispelled "Bearer"
+    {
+        let mut client = server.flight_sql_client("foo").await;
+        client
+            .add_header("authorization", &format!("bearer {TOKEN}"))
+            .unwrap();
+        let error = client.query("SELECT * FROM cpu").await.unwrap_err();
+        assert!(matches!(error, FlightError::Tonic(s) if s.code() == tonic::Code::Unauthenticated));
+    }
+
+    // Invalid token, this actually gives Permission denied
+    {
+        let mut client = server.flight_sql_client("foo").await;
+        client
+            .add_header("authorization", "Bearer invalid-token")
+            .unwrap();
+        let error = client.query("SELECT * FROM cpu").await.unwrap_err();
+        assert!(
+            matches!(error, FlightError::Tonic(s) if s.code() == tonic::Code::PermissionDenied)
+        );
+    }
+
+    // Mispelled header key
+    {
+        let mut client = server.flight_sql_client("foo").await;
+        client
+            .add_header("authorizon", &format!("Bearer {TOKEN}"))
```

Comment:
we should just set this to `auth` to be more clear it's a bad header, much like I did in yesterday's implementation that @jdstrand asked for.